solutions/usaco-833.mdx

```diff
@@ -2,13 +2,106 @@
 id: usaco-833
 source: USACO Bronze 2018 Open
 title: Family Tree
-author: Danh Ta Chi Thanh
+author: Danh Ta Chi Thanh, Ryan Chou
 ---
 
 ## Solution
 
 [Official Analysis (C++)](http://www.usaco.org/current/data/sol_family_bronze_open18.html)
 
+By finding the closest common ancestor and the distances from the ancestor to the two cows, we can uniquely identify their relationship.
+## Implementation
+**Time Complexity:** $\mathcal{O}(N^3)$
+
+
+<LanguageSection>
+<PySection>
+	
+```py
+import sys
+
+sys.stdin = open('family.in', 'r')
+sys.stdout = open('family.out', 'w')
+
+n, a, b = input().split()
+relations = []
+
+for i in range(int(n)):     
+	relations.append(input().split())
+
+# gets the mother of the cow (if it exists)
+def mother(cow):
+	for i in relations:
+		if i[1] == cow:
+			return i[0]
+	return ''
```

Comment:
maybe instead of an empty string return None instead?

Reply:
sure!